packages/vscode-dependencies-validation/src/extension.ts

```diff
@@ -0,0 +1,51 @@
+import { getDependencyIssues } from "@sap-devx/npm-dependencies-validation";
+
+import { DependencyIssue } from "@sap-devx/npm-dependencies-validation/dist/src/types";
+
+import { Uri, ExtensionContext, workspace, window } from "vscode";
+
+export function activate(context: ExtensionContext) {
+  void findIssues();
+
+  void addFileWatcher();
+}
+
+async function findIssues(): Promise<void> {
+  const packageJsonUris: Uri[] = await workspace.findFiles(
+    "package.json",
+    "**​/node_modules/**"
+  );
+  packageJsonUris.forEach((packageJsonUri) => {
+    void displayProblematicDependencies(packageJsonUri);
+  });
+}
+
+function addFileWatcher(): void {
+  const fileWatcher = workspace.createFileSystemWatcher("**/package.json");
+  fileWatcher.onDidChange((uri: Uri) => {
+    void displayProblematicDependencies(uri);
+  });
+
+  fileWatcher.onDidCreate((uri: Uri) => {
+    void displayProblematicDependencies(uri);
+  });
+
+  fileWatcher.onDidDelete((uri: Uri) => {
+    //TODO: check if we need it ??
+    void displayProblematicDependencies(uri);
```

Comment:
check what happens when a file with diagnostics errors get deleted in vscode

Reply:
it is removed from problems view
I will check whether we need to handle this type of action